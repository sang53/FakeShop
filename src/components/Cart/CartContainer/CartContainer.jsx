import classes from "./CartContainer.module.css";
import { use } from "react";
import {
  CartItems,
  CartReducer,
} from "../../../contexts/CartContext/CartContext";

import CartProduct from "../CartProduct/CartProduct";

export default function CartContainer() {
  const output = Object.entries(use(CartItems)).map(([id, quantity]) => {
    return <CartProduct key={id} id={+id} quantity={quantity} />;
  });
  const cartDispatch = use(CartReducer);
  return (
    <div className={classes.cartContainer}>
      <h1>Shopping Cart:</h1>
      {output.length ? output : <div>Your Cart is Empty!</div>}
      <div>
        <button onClick={() => cartDispatch({ type: "reset" })}>
          Remove All
        </button>
        <button>Checkout All</button>
      </div>
    </div>
  );
}
