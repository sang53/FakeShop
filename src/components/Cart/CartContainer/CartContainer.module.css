.cartContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 2rem;
  padding: 1rem;
  min-width: 100%;

  > h1 {
    align-self: center;
  }
}
