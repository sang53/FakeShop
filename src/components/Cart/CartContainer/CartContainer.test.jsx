import { render, screen } from "@testing-library/react";
import { describe, expect, test, vi } from "vitest";
import {
  CartItems,
  CartReducer,
} from "../../../contexts/CartContext/CartContext";
import CartContainer from "./CartContainer";

vi.mock("../CartProduct/CartProduct", () => ({
  default: ({ id, quantity }) => (
    <div>
      id: {id}; quantity:{quantity}
    </div>
  ),
}));

const mockCart = { 1: 2, 6: 4 };

function renderCartContainer(cartValue) {
  const mockDispatch = vi.fn();
  const renderReturns = render(
    <CartItems value={cartValue}>
      <CartReducer value={mockDispatch}>
        <CartContainer />
      </CartReducer>
    </CartItems>
  );
  return { ...renderReturns, mockDispatch };
}

describe("snapshot tests", () => {
  test("empty cart", () => {
    const { container } = renderCartContainer({});
    expect(container).toMatchSnapshot();
  });

  test("cart with products", () => {
    const { container } = renderCartContainer(mockCart);
    expect(container).toMatchSnapshot();
  });
});

test("remove all button", async () => {
  const { mockDispatch } = renderCartContainer(mockCart);
  await user.click(screen.getByRole("button", { name: /remove/i }));
  expect(mockDispatch).toHaveBeenCalledOnce();
  expect(mockDispatch.mock.calls[0][0]).toMatchInlineSnapshot(`
    {
      "type": "reset",
    }
  `);
});
