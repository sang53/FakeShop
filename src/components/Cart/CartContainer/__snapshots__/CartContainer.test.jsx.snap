// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot tests > cart with products 1`] = `
<div>
  <div
    class="_cartContainer_6fd72c"
  >
    <h1>
      Shopping Cart:
    </h1>
    <div>
      id: 
      1
      ; quantity:
      2
    </div>
    <div>
      id: 
      6
      ; quantity:
      4
    </div>
    <div>
      <button>
        Remove All
      </button>
      <button>
        Checkout All
      </button>
    </div>
  </div>
</div>
`;

exports[`snapshot tests > empty cart 1`] = `
<div>
  <div
    class="_cartContainer_6fd72c"
  >
    <h1>
      Shopping Cart:
    </h1>
    <div>
      Your Cart is Empty!
    </div>
    <div>
      <button>
        Remove All
      </button>
      <button>
        Checkout All
      </button>
    </div>
  </div>
</div>
`;
